app/rpc/types/utils.go

```diff
@@ -238,3 +243,31 @@ func GetBlockCumulativeGas(cdc *codec.Codec, block *tmtypes.Block, idx int) uint
 	}
 	return gasUsed
 }
+
+// EthHeaderWithBlockHashFromTendermint gets the eth Header with block hash from Tendermint block inside
+func EthHeaderWithBlockHashFromTendermint(tmHeader *tmtypes.Header, cli rpcclient.Client) (header *HeaderWithBlockHash, err error) {
+	if tmHeader == nil {
+		return header, errors.New("failed. nil tendermint block header")
+	}
+
+	tmBlock, err := cli.Block(&tmHeader.Height)
```

Comment:
it is not a good idea to make a query request in websocket.
it could use common.BytesToHash(tmHeader) to calculate hash